stock_ownership_availability_rules/model/stock_account.py

```diff
@@ -0,0 +1,122 @@
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) Rooms For (Hong Kong) Limited T/A OSCG
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import osv
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class stock_quant(osv.osv):
+    _inherit = "stock.quant"
+
+    def _get_internal_loc(self, cr, uid, move, context=None):
+        if move.location_dest_id.usage == 'internal':
+            return move.location_dest_id.id
+        elif move.location_id.usage == 'internal':
+            return move.location_id.id
+    
+    """
+    Following method is a copy from the official Odoo branch
+    (addons/stock_account/stock_account.py) with only change on the condition
+    where judgment on quant's owner is made (see lines with comment
+    "REPLACED").
+    This is due to the fact that this module makes the owner_id a required
+    field regardless of who owns the stock, while standard Odoo assumes the
+    field should be blank in case the stock is owned by the company.
+    As this overrides the original method, the content may need to be
+    backported from the official branch as necessary.
+    """
+    def _account_entry_move(self, cr, uid, quants, move, context=None):
+        """
+        Accounting Valuation Entries
+
+        quants: browse record list of Quants to create accounting valuation
+            entries for. Unempty and all quants are supposed to have the same
+            location id (thay already moved in)
+        move: Move to use. browse record
+        """
+        if context is None:
+            context = {}
+        location_obj = self.pool.get('stock.location')
+        location_from = move.location_id
+        location_to = quants[0].location_id
+        company_from = location_obj._location_owner(cr, uid, location_from,
+            context=context)
+        company_to = location_obj._location_owner(cr, uid, location_to,
+            context=context)
+
+        if move.product_id.valuation != 'real_time':
+            return False
+        for q in quants:
+            # if q.owner_id:  # REPLACED FROM
+            if q.owner_id and q.owner_id.id != \
```

Comment:
Now that I think of it, owner_id is now required, so the first part is always true.